### DISCUSSION TITLE
Pandoc now supports djot

### DISCUSSION THREAD

**Discussion by jgm** [jgm/djot#270] [Announcements]
both as an input and an output format. <https://github.com/jgm/pandoc/releases/tag/3.1.12>

**Comment by dai** [comment]
Finally, Pandoc supports Djot!

Big shout it out Professer.

**Comment by uvtc** [comment]
Thank you!

**Comment by Omikhleia** [comment]
Erm... What about round-tripping?
(Edit: walking on eggs...)

**Reply by jgm** [reply]
The djoths library does do round-trip tests for all of the test cases.  I caught quite a few issues in the writer this way.  Caveat: currently this is only tested without automatic line-wrapping, which may introduce issues we haven't caught yet.

**Reply by Omikhleia** [reply]
I was actually thinking to round-tripping conversions between formats with same base features (A --> B --> A),  such as (running pandoc 3.1.13):

```
$ pandoc -f markdown+mark -t djot
==toto==
[toto]{.mark}
$ pandoc -f djot -t markdown+mark
{=toto=}
[toto]{.highlighted}
```

Or

```
$ pandoc -f djot -t markdown+strikeout
{-toto-}
[toto]{.deleted}
$ pandoc -f markdown+strikeout -t djot
~~toto~~
{-toto-}
```

This does not seem as consistent as e.g.

```
$ pandoc -f html -t markdown+strikeout
<p><del>toto</del></p>
~~toto~~
$pandoc -f html -t markdown+mark
<p><mark>mark</mark></p>
==mark==
```

But there might be more that meets the eye there, too:

```
$ pandoc -f djot -t html
{-toto-} {=toto=}
<p><span class="deleted">toto</span> <span class="highlighted">toto</span></p>
$ pandoc -f markdown+mark+strikeout -t html
~~toto~~ ==toto==
<p><del>toto</del> <mark>toto</mark></p>
```

**Reply by Omikhleia** [reply]
Or

```
$ pandoc -f html -t djot
<p><del>toto</del> <ins>toto</ins></p>
{-toto-} [toto]{.underline}
$ pandoc -t html -f djot
{-toto-} {+toto+}
<p><span class="deleted">toto</span> <span
class="inserted">toto</span></p>
```
(non-symmetric mapping, and underline vs. inserted)

**Reply by jgm** [reply]
To understand what is going on, you'll want to look at how pandoc is parsing things: instead of djot -> X -> djot, do:

- djot -> native
- native -> X
- X -> native
- native -> djot

This will tell you where things are changing. Perhaps a concrete suggestion for specific changes might come out of this, but it's delicate -- pandoc has its own concepts and they differ a bit from djot's.

**Reply by Omikhleia** [reply]
What is the rationale for parsing `{=xxx=}` into an AST Span with `.highlighted`:
https://github.com/jgm/pandoc/blob/main/src/Text/Pandoc/Readers/Djot.hs#L192

Rather than `.mark`, which is what the Markdown reader does for the (seemingly) semantically identical `==xxx==`:
https://github.com/jgm/pandoc/blob/edf21e6dab8cddf15dac6f494291396d4f4df8db/src/Text/Pandoc/Readers/Markdown.hs#L1726

And which is supported for rendering as `<mark>` in the HTML writer, around those lines it seems...
https://github.com/jgm/pandoc/blob/edf21e6dab8cddf15dac6f494291396d4f4df8db/src/Text/Pandoc/Writers/HTML.hs#L1402

... Or by the LaTeX Write as `\hl`:
https://github.com/jgm/pandoc/blob/edf21e6dab8cddf15dac6f494291396d4f4df8db/src/Text/Pandoc/Writers/LaTeX.hs#L822-L824

Or is `{=djot=}` considered different from extended `==markdown==` for any reason?

**Comment by bpj** [comment]
@jgm what is the status of this? Will it be in the next release of Pandoc?

**Reply by jgm** [reply]
It's already in the current release.
